tilelang/tools/lower_trace/core.py

```
@@ -0,0 +1,1249 @@
+
+    Prefers the newer ``inspect_source``; falls back to the legacy ``get_source``
+    shipped with ``3rdparty/tvm``. Returns ``""`` when neither is available so
+    downstream string handling stays safe.
+    """
+    for _attr in ("inspect_source", "get_source"):
+        _fn = getattr(mod, _attr, None)
+        if callable(_fn):
+            try:
+                return _fn() or ""
+            except Exception:
+                continue
```

Ruff (0.15.18)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)